app/src/main/java/nu/yona/app/ui/dashboard/WeekActivityDetailFragment.java

```diff
@@ -261,32 +273,59 @@ public void onDestroyView() {
+       getCurrentWeekActivityDetails(activity);
+    }
+
 
-            for (int i = embeddedYonaActivity.getWeekActivityList().size() - 1; i >= 0; i--) {
-                try {
+    public void getCurrentWeekActivityDetails(WeekActivity weekActivity){
+
+        YonaActivity.getActivity().showLoadingView(true, null);
+        if (!isDataLoading) {
+            isDataLoading = true;
+            DataLoadListenerImpl dataLoadListenerImpl =  new DataLoadListenerImpl(((result) -> handleWeekActivityDetailsFetchSuccess((WeekActivity) result)), ((result) ->handleWeekActivityDetailsFetchFailure(result)),null);
+            APIManager.getInstance().getActivityManager().getDetailOfEachWeekSpreadWithWeekActivity(weekActivity, dataLoadListenerImpl);
+        }
+    }
+
+    private  Object handleWeekActivityDetailsFetchSuccess(WeekActivity result) {
+        try {
+            YonaActivity.getActivity().showLoadingView(false, null);
+            isDataLoading = false;
+            activity = (WeekActivity) result;
+            weekActivityList = new ArrayList<>();
+            EmbeddedYonaActivity embeddedYonaActivity = YonaApplication.getEventChangeManager().getDataState().getEmbeddedWeekActivity();
+            if (embeddedYonaActivity != null && embeddedYonaActivity.getWeekActivityList() != null && embeddedYonaActivity.getWeekActivityList().size() > 0) {
+                for (int i = embeddedYonaActivity.getWeekActivityList().size() - 1; i >= 0; i--) {
                     if (embeddedYonaActivity.getWeekActivityList().get(i).getYonaGoal().getLinks().getSelf().getHref().equals(activity.getLinks().getYonaGoal().getHref())) {
                         weekActivityList.add(embeddedYonaActivity.getWeekActivityList().get(i));
                     }
-                } catch (Exception e) {
-                    AppUtils.throwException(WeekActivityDetailFragment.class.getSimpleName(), e, Thread.currentThread(), null);
                 }
-            }
-            int itemIndex = getIndex(activity);
-            if (itemIndex >= 0) {
-                customPageAdapter.notifyDataSetChanged(weekActivityList);
-                fetchComments(itemIndex);
-                viewPager.setCurrentItem(itemIndex);
-                updateFlow(itemIndex);
+                int itemIndex = getIndex(activity);
+                if (itemIndex >= 0) {
+                    customPageAdapter.notifyDataSetChanged(weekActivityList);
+                    fetchComments(itemIndex);
+                    if (itemIndex != viewPager.getCurrentItem()) {
+                        viewPager.setCurrentItem(itemIndex);
+                    }
+                    updateFlow(itemIndex);
+                } else {
+                    goBack();
+                }
             } else {
                 goBack();
             }
-        } else {
-            goBack();
+            setDayDetailTitleAndIcon();
+        } catch (NullPointerException e) {
```

Comment:
Catching an NPE is a code smell. Why is it necessary here?

Reply:
We have removed quite a few null checks in the inner classes as part of re-factoring. So I thought of adding it to avoid the crashes.

Reply:
While correcting code with your earlier reviews, I have removed (embeddedYonaActivity != null) check which caused the crash on launch. So added it to avoid the crash. We can remove it now as code is safe with null check.